### ISSUE TITLE
Crash with store.scope

### ISSUE THREAD

**Issue by natemann** [pointfreeco/swift-composable-architecture#228]
**Describe the bug**
This bug only occurs on actual devices, not on simulators. I have attached a simple project to reproduce. Simply keep pressing the 'add text' button, and text, then dismiss. after 3-4 times, the app will crash. I have also tested this behavior with vanilla swiftUI, and it does not crash. From my debugging, it seems to be happening in the store.scope method, but I cannot narrow it down any more

**To Reproduce**
[ScopeCrash.zip](https://github.com/pointfreeco/swift-composable-architecture/files/4939916/ScopeCrash.zip)
**Environment**
 - Xcode [e.g. 11.3.1]
 - Swift [e.g. 5.2]
 - OS (if applicable): [e.g. iOS 13]

**Comment by stephencelis** [comment]
Hey @natemann! Thanks for reporting. I can't seem to reproduce. Can you clarify a few things?

> Simply keep pressing the 'add text' button, and text, then dismiss. after 3-4 times, the app will crash.

Are you reproducing by tapping "Add text", editing the field with some text, and then tapping "Add" to dismiss? I seem to be able to do this a bunch of times without seeing any crashing.

I did notice that dragging to dismiss stop the "Add text" button from working, but it doesn't crash the app.

Any chance you can post a screen shot of where in `scope` it crashes?

**Comment by natemann** [comment]
@stephencelis Yes, I press the add text button, edit text, then press the add button. It always crashes after 3 times for me. I did it on an iPhone 11 Pro running 13.5.1.

I am not sure if Scope is the cause, or even it Composable Architecture is at fault. That is why I added the demo in vanilla SwiftUI. 

<img width="1392" alt="Screen Shot 2020-07-17 at 4 24 50 PM" src="https://user-images.githubusercontent.com/5096440/87828095-3b861700-c84a-11ea-82e7-94b1647a2e20.png">
